### ISSUE TITLE
XcodeDefault.xctoolchain/usr/bin/ranlib: file: libsolidity.a(AsmAnalysisInfo.cpp.o) has no symbols

### ISSUE THREAD

**Issue by avatar-lavventura** [ethereum/homebrew-ethereum#128]
I am trying to install latest `solc` compiler. I keep getting this error, I am not sure how to fix this: 
Thank you for your valuable time and help.
```
solidity-0.4.10$ ./scripts/install_deps.sh
Installing solidity dependencies on macOS 10.12 Sierra.
Updated 1 tap (homebrew/core).
No changes to formulae.
Warning: boost 1.65.1 is already installed
Warning: cmake 3.9.3_1 is already installed
./scripts/install_deps.sh: line 99: [: missing `]'
==> Upgrading 1 outdated package, with result:
ethereum/ethereum/solidity 0.4.17
==> Upgrading ethereum/ethereum/solidity
==> Downloading https://github.com/ethereum/solidity/releases/download/v0.4.17/solidity_0.4.17.tar.gz
Already downloaded: /Users/alper/Library/Caches/Homebrew/solidity-0.4.17.tar.gz
==> cmake . -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/solidity/0.4.17 -DCMAKE_BUILD_TYPE=Release -
==> make install
Last 15 lines from /Users/alper/Library/Logs/Homebrew/solidity/02.make:
[ 62%] Building CXX object libsolidity/CMakeFiles/solidity.dir/parsing/Token.cpp.o
cd /tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/libsolidity && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -I/tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17 -I/tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/include -isystem /usr/local/include -isystem /tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/deps/include  -std=c++11 -fpermissive -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.12   -fstack-protector-strong -Wall -Wextra -Werror -Wno-unknown-pragmas -Wno-unused-function -Wno-dangling-else -o CMakeFiles/solidity.dir/parsing/Token.cpp.o -c /tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/libsolidity/parsing/Token.cpp
[ 63%] Building CXX object libsolidity/CMakeFiles/solidity.dir/__/libjulia/backends/evm/EVMAssembly.cpp.o
cd /tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/libsolidity && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -I/tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17 -I/tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/include -isystem /usr/local/include -isystem /tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/deps/include  -std=c++11 -fpermissive -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.12   -fstack-protector-strong -Wall -Wextra -Werror -Wno-unknown-pragmas -Wno-unused-function -Wno-dangling-else -o CMakeFiles/solidity.dir/__/libjulia/backends/evm/EVMAssembly.cpp.o -c /tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/libjulia/backends/evm/EVMAssembly.cpp
[ 64%] Building CXX object libsolidity/CMakeFiles/solidity.dir/__/libjulia/backends/evm/EVMCodeTransform.cpp.o
cd /tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/libsolidity && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -I/tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17 -I/tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/include -isystem /usr/local/include -isystem /tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/deps/include  -std=c++11 -fpermissive -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.12   -fstack-protector-strong -Wall -Wextra -Werror -Wno-unknown-pragmas -Wno-unused-function -Wno-dangling-else -o CMakeFiles/solidity.dir/__/libjulia/backends/evm/EVMCodeTransform.cpp.o -c /tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/libjulia/backends/evm/EVMCodeTransform.cpp
[ 64%] Linking CXX static library libsolidity.a
cd /tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/libsolidity && /usr/local/Cellar/cmake/3.9.3_1/bin/cmake -P CMakeFiles/solidity.dir/cmake_clean_target.cmake
cd /tmp/solidity-20170923-34560-kfkbkk/solidity_0.4.17/libsolidity && /usr/local/Cellar/cmake/3.9.3_1/bin/cmake -E cmake_link_script CMakeFiles/solidity.dir/link.txt --verbose=1
/usr/bin/ar qc libsolidity.a  CMakeFiles/solidity.dir/analysis/ConstantEvaluator.cpp.o CMakeFiles/solidity.dir/analysis/DeclarationContainer.cpp.o CMakeFiles/solidity.dir/analysis/DocStringAnalyser.cpp.o CMakeFiles/solidity.dir/analysis/GlobalContext.cpp.o CMakeFiles/solidity.dir/analysis/NameAndTypeResolver.cpp.o CMakeFiles/solidity.dir/analysis/PostTypeChecker.cpp.o CMakeFiles/solidity.dir/analysis/ReferencesResolver.cpp.o CMakeFiles/solidity.dir/analysis/SemVerHandler.cpp.o CMakeFiles/solidity.dir/analysis/StaticAnalyzer.cpp.o CMakeFiles/solidity.dir/analysis/SyntaxChecker.cpp.o CMakeFiles/solidity.dir/analysis/TypeChecker.cpp.o CMakeFiles/solidity.dir/analysis/ViewPureChecker.cpp.o CMakeFiles/solidity.dir/ast/AST.cpp.o CMakeFiles/solidity.dir/ast/ASTAnnotations.cpp.o CMakeFiles/solidity.dir/ast/ASTJsonConverter.cpp.o CMakeFiles/solidity.dir/ast/ASTPrinter.cpp.o CMakeFiles/solidity.dir/ast/ASTUtils.cpp.o CMakeFiles/solidity.dir/ast/Types.cpp.o CMakeFiles/solidity.dir/codegen/ABIFunctions.cpp.o CMakeFiles/solidity.dir/codegen/ArrayUtils.cpp.o CMakeFiles/solidity.dir/codegen/Compiler.cpp.o CMakeFiles/solidity.dir/codegen/CompilerContext.cpp.o CMakeFiles/solidity.dir/codegen/CompilerUtils.cpp.o CMakeFiles/solidity.dir/codegen/ContractCompiler.cpp.o CMakeFiles/solidity.dir/codegen/ExpressionCompiler.cpp.o CMakeFiles/solidity.dir/codegen/LValue.cpp.o CMakeFiles/solidity.dir/formal/SMTChecker.cpp.o CMakeFiles/solidity.dir/formal/SMTLib2Interface.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmAnalysis.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmAnalysisInfo.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmCodeGen.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmParser.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmPrinter.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmScope.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmScopeFiller.cpp.o CMakeFiles/solidity.dir/interface/ABI.cpp.o CMakeFiles/solidity.dir/interface/AssemblyStack.cpp.o CMakeFiles/solidity.dir/interface/CompilerStack.cpp.o CMakeFiles/solidity.dir/interface/ErrorReporter.cpp.o CMakeFiles/solidity.dir/interface/Exceptions.cpp.o CMakeFiles/solidity.dir/interface/GasEstimator.cpp.o CMakeFiles/solidity.dir/interface/Natspec.cpp.o CMakeFiles/solidity.dir/interface/SourceReferenceFormatter.cpp.o CMakeFiles/solidity.dir/interface/StandardCompiler.cpp.o CMakeFiles/solidity.dir/interface/Version.cpp.o CMakeFiles/solidity.dir/parsing/DocStringParser.cpp.o CMakeFiles/solidity.dir/parsing/Parser.cpp.o CMakeFiles/solidity.dir/parsing/ParserBase.cpp.o CMakeFiles/solidity.dir/parsing/Scanner.cpp.o CMakeFiles/solidity.dir/parsing/Token.cpp.o CMakeFiles/solidity.dir/__/libjulia/backends/evm/EVMAssembly.cpp.o CMakeFiles/solidity.dir/__/libjulia/backends/evm/EVMCodeTransform.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libsolidity.a(AsmAnalysisInfo.cpp.o) has no symbols
/usr/bin/ranlib libsolidity.a
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libsolidity.a(AsmAnalysisInfo.cpp.o) has no symbols
[ 64%] Built target solidity
make: *** [all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/ethereum/homebrew-ethereum/issues

These open issues may also help:
Add a dependency on jsoncpp for solidity? https://github.com/ethereum/homebrew-ethereum/issues/123
Error when installing Solidity compiler https://github.com/ethereum/homebrew-ethereum/issues/96
Add ccache dependency and adds a bottle https://github.com/ethereum/homebrew-ethereum/pull/119

Error: You have MacPorts or Fink installed:
  /sw/bin/fink

This can cause trouble. You don't have to uninstall them, but you may want to
temporarily move them out of the way, e.g.

  sudo mv /opt/local ~/macports
```

**Comment by godsflaw** [comment]
bump, I'm getting this too for `ethereum/ethereum/solidity 0.4.17`.

`Xcode Version 9.0 (9A235)`
```
% brew --version
Homebrew 1.3.4
Homebrew/homebrew-core (git revision d735c; last commit 2017-09-25)
```


```
Updating Homebrew...
==> Upgrading 1 outdated package, with result:
ethereum/ethereum/solidity 0.4.17
==> Upgrading ethereum/ethereum/solidity 
==> Downloading https://github.com/ethereum/solidity/releases/download/v0.4.17/solidity_0.4.17.tar.gz
Already downloaded: /Users/godsflaw/Library/Caches/Homebrew/solidity-0.4.17.tar.gz
==> cmake . -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/u
==> make install

Last 15 lines from /Users/godsflaw/Library/Logs/Homebrew/solidity/02.make:
[ 62%] Building CXX object libsolidity/CMakeFiles/solidity.dir/parsing/Token.cpp.o
cd /tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/libsolidity && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -I/tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17 -I/tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/include -isystem /usr/local/include -isystem /tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/deps/include  -std=c++11 -fpermissive -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.12   -fstack-protector-strong -Wall -Wextra -Werror -Wno-unknown-pragmas -Wno-unused-function -Wno-dangling-else -o CMakeFiles/solidity.dir/parsing/Token.cpp.o -c /tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/libsolidity/parsing/Token.cpp
[ 63%] Building CXX object libsolidity/CMakeFiles/solidity.dir/__/libjulia/backends/evm/EVMAssembly.cpp.o
cd /tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/libsolidity && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -I/tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17 -I/tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/include -isystem /usr/local/include -isystem /tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/deps/include  -std=c++11 -fpermissive -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.12   -fstack-protector-strong -Wall -Wextra -Werror -Wno-unknown-pragmas -Wno-unused-function -Wno-dangling-else -o CMakeFiles/solidity.dir/__/libjulia/backends/evm/EVMAssembly.cpp.o -c /tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/libjulia/backends/evm/EVMAssembly.cpp
[ 64%] Building CXX object libsolidity/CMakeFiles/solidity.dir/__/libjulia/backends/evm/EVMCodeTransform.cpp.o
cd /tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/libsolidity && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -I/tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17 -I/tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/include -isystem /usr/local/include -isystem /tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/deps/include  -std=c++11 -fpermissive -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.12   -fstack-protector-strong -Wall -Wextra -Werror -Wno-unknown-pragmas -Wno-unused-function -Wno-dangling-else -o CMakeFiles/solidity.dir/__/libjulia/backends/evm/EVMCodeTransform.cpp.o -c /tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/libjulia/backends/evm/EVMCodeTransform.cpp
[ 64%] Linking CXX static library libsolidity.a
cd /tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/libsolidity && /usr/local/Cellar/cmake/3.9.3_1/bin/cmake -P CMakeFiles/solidity.dir/cmake_clean_target.cmake
cd /tmp/solidity-20170925-53772-12wmhb/solidity_0.4.17/libsolidity && /usr/local/Cellar/cmake/3.9.3_1/bin/cmake -E cmake_link_script CMakeFiles/solidity.dir/link.txt --verbose=1
/usr/bin/ar qc libsolidity.a  CMakeFiles/solidity.dir/analysis/ConstantEvaluator.cpp.o CMakeFiles/solidity.dir/analysis/DeclarationContainer.cpp.o CMakeFiles/solidity.dir/analysis/DocStringAnalyser.cpp.o CMakeFiles/solidity.dir/analysis/GlobalContext.cpp.o CMakeFiles/solidity.dir/analysis/NameAndTypeResolver.cpp.o CMakeFiles/solidity.dir/analysis/PostTypeChecker.cpp.o CMakeFiles/solidity.dir/analysis/ReferencesResolver.cpp.o CMakeFiles/solidity.dir/analysis/SemVerHandler.cpp.o CMakeFiles/solidity.dir/analysis/StaticAnalyzer.cpp.o CMakeFiles/solidity.dir/analysis/SyntaxChecker.cpp.o CMakeFiles/solidity.dir/analysis/TypeChecker.cpp.o CMakeFiles/solidity.dir/analysis/ViewPureChecker.cpp.o CMakeFiles/solidity.dir/ast/AST.cpp.o CMakeFiles/solidity.dir/ast/ASTAnnotations.cpp.o CMakeFiles/solidity.dir/ast/ASTJsonConverter.cpp.o CMakeFiles/solidity.dir/ast/ASTPrinter.cpp.o CMakeFiles/solidity.dir/ast/ASTUtils.cpp.o CMakeFiles/solidity.dir/ast/Types.cpp.o CMakeFiles/solidity.dir/codegen/ABIFunctions.cpp.o CMakeFiles/solidity.dir/codegen/ArrayUtils.cpp.o CMakeFiles/solidity.dir/codegen/Compiler.cpp.o CMakeFiles/solidity.dir/codegen/CompilerContext.cpp.o CMakeFiles/solidity.dir/codegen/CompilerUtils.cpp.o CMakeFiles/solidity.dir/codegen/ContractCompiler.cpp.o CMakeFiles/solidity.dir/codegen/ExpressionCompiler.cpp.o CMakeFiles/solidity.dir/codegen/LValue.cpp.o CMakeFiles/solidity.dir/formal/SMTChecker.cpp.o CMakeFiles/solidity.dir/formal/SMTLib2Interface.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmAnalysis.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmAnalysisInfo.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmCodeGen.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmParser.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmPrinter.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmScope.cpp.o CMakeFiles/solidity.dir/inlineasm/AsmScopeFiller.cpp.o CMakeFiles/solidity.dir/interface/ABI.cpp.o CMakeFiles/solidity.dir/interface/AssemblyStack.cpp.o CMakeFiles/solidity.dir/interface/CompilerStack.cpp.o CMakeFiles/solidity.dir/interface/ErrorReporter.cpp.o CMakeFiles/solidity.dir/interface/Exceptions.cpp.o CMakeFiles/solidity.dir/interface/GasEstimator.cpp.o CMakeFiles/solidity.dir/interface/Natspec.cpp.o CMakeFiles/solidity.dir/interface/SourceReferenceFormatter.cpp.o CMakeFiles/solidity.dir/interface/StandardCompiler.cpp.o CMakeFiles/solidity.dir/interface/Version.cpp.o CMakeFiles/solidity.dir/parsing/DocStringParser.cpp.o CMakeFiles/solidity.dir/parsing/Parser.cpp.o CMakeFiles/solidity.dir/parsing/ParserBase.cpp.o CMakeFiles/solidity.dir/parsing/Scanner.cpp.o CMakeFiles/solidity.dir/parsing/Token.cpp.o CMakeFiles/solidity.dir/__/libjulia/backends/evm/EVMAssembly.cpp.o CMakeFiles/solidity.dir/__/libjulia/backends/evm/EVMCodeTransform.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libsolidity.a(AsmAnalysisInfo.cpp.o) has no symbols
/usr/bin/ranlib libsolidity.a
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libsolidity.a(AsmAnalysisInfo.cpp.o) has no symbols
[ 64%] Built target solidity
make: *** [all] Error 2
```
I'll dig more in a bit, but for now I just wanted to flag this as more widespread.

**Comment by erkmos** [comment]
```
Undefined symbols for architecture x86_64:
  "Json::Value::append(Json::Value&&)", referenced from:
      dev::eth::Assembly::assemblyJSON(std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > > const&) const in libevmasm.a(Assembly.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Guess that's the problem; `lllc` build fails (jsoncpp doesn't get linked properly?) Tried unlinking my `jsoncpp` (`brew unlink jsoncpp`) and it builds fine. I think solidity uses an older version of `jsoncpp` with a different API which it bundles with the project. Not really sure how to fix this with homebrew though, maybe it needs to be fixed in solidity?

**Comment by axic** [comment]
This issue is tracked in https://github.com/ethereum/solidity/issues/2956

**Comment by godsflaw** [comment]
Seems like @chfast is on this one at the solidity level.  So that folks don't block on this problem until there is a fix, solidity is attempting to do a static link against a locally package library, and it's pulling in the system library instead.  Which means the workaround is easy, all you need to do is:
```
% brew unlink jsoncpp
% brew upgrade solidity
% brew link jsoncpp
% solc --version
solc, the solidity compiler commandline interface
Version: 0.4.17+commit.bdeb9e52.Darwin.appleclang
```
Obviously, if you are trying to install for the first time, use `brew install solidity` instead of `brew  upgrade solidity`.

I hope this helps anyone blocked on this until there is a fix.

**Comment by avatar-lavventura** [comment]
Thank you @godsflaw great solution. It works!!!!

**Comment by axic** [comment]
This should be solved by https://github.com/ethereum/solidity/issues/2956. Please open a new issue on https://github.com/ethereum/solidity if the problem persists with the 0.4.20 release.
